operations/observability/Observability.md

```diff
@@ -0,0 +1,318 @@
+  - Product stream health
+
+- Drill down health dashboard with roll up.
+  - Aggregate checks - Problem/OK counts across each product stream.
+  - Functional checks - Product stream functional Goss suites for key areas.
+    For example, REST APIs, micro-services, Kubernetes, network, and database health for the specific product stream.
+  - Granular checks - Individual Goss tests for component level health checks within a functional area of a given product stream.
+    For example, management switch configuration verification, routing table checks on OS, gateway tests, and Container Network Interface (CNI) tests.
+
+- Boot, install, and upgrade duration monitoring.
+  This does automatic calculation and reporting of both the time a given section of install/upgrade has taken, as well as metrics on Goss test successes and failures.
+
+- Time, node, product stream, capacity, and other dimensions based health and performance insight.
+
+- Multi-interval continuous health checks.
+  This defines and implements regular scheduled health checks to occur both during and after the installation and upgrade.
+  We can select the appropriate time interval in the Grafana dashboard to check the results for past six hours, a day, and a week.
+
+- Automate the Observability framework , installation, and configuration of the components involved in the Install and Upgrade Observability Framework in a CSM environment.
+
+- The automatic generation of configurable Grafana dashboards that provide key insights and Key Performance Indicators (KPIs).
+  These dashboards show the frequency of errors across the complex systems, include panels to visualize the outliers, and provide a way to identify the trends in the complex system across different dimensions.
+
+The following IUF topics are discussed in the following subsections.
+
+- [Automation of Observability Framework](#automation-of-observability-framework)
+  - [Features](#observability-framework-features)
+  - [Observability workflow](#observability-workflow)
+  - [`systemd` services](#systemd-services)
+- [IUF timing dashboard](#iuf-timing-dashboard)
+  - [Features](#timing-dashboard-features)
+  - [Prometheus metrics using Argo workflow](#prometheus-metrics-using-argo-workflow)
+  - [Timing dashboard](#timing-dashboard)
+- [Goss tests for PIT and NCN](#goss-tests-for-pit-and-ncn)
+  - [Overview](#goss-test-overview)
+  - [Workflow](#workflow)
+  - [Log file format](#log-file-format)
+  - [Grok-exporter deployment, service, and service-monitor](#grok-exporter-deployment-service-and-service-monitor)
+  - [Configuration file for the grok-exporter](#configuration-file-for-the-grok-exporter)
+  - [Prometheus metrics and Grafana dashboard](#prometheus-metrics-and-grafana-dashboard)
+- [Error dashboard](#error-dashboard)
+  - [Features](#error-dashboards-features)
+  - [Error dashboards](#error-dashboards)
+
+## Automation of Observability Framework
```

Comment:
This first section really seems like it is talking about the Observability Framework on the PIT node. It appears to be explaining that the framework is automatically there on the PIT node, and shows the services that are running as part of it. 

Therefore, I suggest changing the name of the section to make that clearer. Please let me know if I am misunderstanding the intent of this first section.

```suggestion
- [Observability Framework during CSM installs](#observability-framework-during-csm-installs)
  - [Overview](#overview-of-observability-framework-on-pit-node)
  - [Observability workflow](#observability-framework-workflow-on-pit-node)
  - [`systemd` services](#observability-framework-systemd-services-on-pit-node)
- [IUF timing dashboard](#iuf-timing-dashboard)
  - [Features](#timing-dashboard-features)
  - [Prometheus metrics using Argo workflow](#prometheus-metrics-using-argo-workflow)
  - [Timing dashboard](#timing-dashboard)
- [Goss tests for PIT and NCN](#goss-tests-for-pit-and-ncn)
  - [Overview](#goss-test-overview)
  - [Workflow](#workflow)
  - [Log file format](#log-file-format)
  - [Grok-exporter deployment, service, and service-monitor](#grok-exporter-deployment-service-and-service-monitor)
  - [Configuration file for the grok-exporter](#configuration-file-for-the-grok-exporter)
  - [Prometheus metrics and Grafana dashboard](#prometheus-metrics-and-grafana-dashboard)
- [Error dashboard](#error-dashboard)
  - [Features](#error-dashboards-features)
  - [Error dashboards](#error-dashboards)

## Observability Framework during CSM installs
```

Reply:
changed